docs/developer-docs/integrations/sns/managing/making-proposals.md

```diff
@@ -42,6 +42,619 @@ For these cases, SNSs have so called 'generic proposals'. These are custom propo
+                        validator_method_name = opt text;
+                        target_method_name = opt text;
+                    }
+                };
+            }
+        };
+    }
+)'
+```
+
+## `RemoveGenericNervousSystemFunction`
+
+### Relevant type signatures
+
+```candid
+    type RemoveGenericNervousSystemFunction : nat64;
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "SNS Launch";
+        url = "https://dfinity.org";
+        summary = "A motion to start the SNS";
+        action = opt variant {
+            RemoveGenericNervousSystemFunction = 123456789:nat64;
+        };
+    }
+)'
+```
+
+## `UpgradeSnsToNextVersion`
+
+### Relevant type signatures
+
+```candid
+    type UpgradeSnsToNextVersion : record {};
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "SNS Launch";
+        url = "https://dfinity.org";
+        summary = "A motion to start the SNS";
+        action = opt variant {
+            UpgradeSnsToNextVersion = record {};
+        };
+    }
+)'
+```
+
+## `RegisterDappCanisters`
+
+### Relevant type signatures
+
+```candid
+    type RegisterDappCanisters : RegisterDappCanisters;
+
+    type RegisterDappCanisters = record { canister_ids : vec principal };
+```
+
+### Putting it together
+
+```candid
+    type RegisterDappCanisters: record {
+        canister_ids : vec principal
+    };
+```
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "SNS Launch";
+        url = "https://dfinity.org";
+        summary = "A motion to start the SNS";
+        action = opt variant {
+            RegisterDappCanisters = record {
+                canister_ids = vec {principal "ltyfs-qiaaa-aaaak-aan3a-cai", principal "ltyfs-qiaaa-aaaak-aan3a-cai"};
+            };
+        };
+    }
+)'
+```
+
+## `TransferSnsTreasuryFunds`
+
+### Relevant type signatures
+
+```candid
+    type TransferSnsTreasuryFunds = record {
+        from_treasury : int32;
+        to_principal : opt principal;
+        to_subaccount : opt Subaccount;
+        memo : opt nat64;
+        amount_e8s : nat64;
+    };
+
+    type Subaccount = record { subaccount : vec nat8 };
+```
+
+### Putting it together
+
+```candid
+    type TransferSnsTreasuryFunds = record {
+        from_treasury : int32;
+        to_principal : opt principal;
+        to_subaccount : opt record { subaccount : vec nat8 };
+        memo : opt nat64;
+        amount_e8s : nat64;
+    };
+```
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "Lorem ipsum";
+        url = "Lorem ipsum";
+        summary = "Lorem ipsum";
+        action = opt variant {
+            TransferSnsTreasuryFunds = record {
+                from_treasury = int32;
```

Comment:
would fill in a concrete example, e.g., to move sth from the ICP treasury to a concrete subaccount and explain in the example taht this is what this does and why